topsort-api-v2.yml

```diff
@@ -466,6 +497,236 @@ components:
+            If there is no click model on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: b39d39ed-ea0e-4059-9d15-4990b39c85a2
+    Purchase:
+      description: >
+        A purchase is sent to Topsort once the user makes a payment transaction and it has been confirmed by the payment provider or financial entity in charge processing the transaction.
+      type: object
+      required:
+        - occurredAt
+        - opaqueUserId
+        - items
+        - id
+      properties:
+        occurredAt:
+          type: string
+          description: RFC3339 formatted timestamp including UTC offset
+          example: '2021-10-12T07:20:50.52Z'
+          format: date-time
+        opaqueUserId:
+          type: string
+          description: The opaque user ID allows correlating user activity during a session whether or not they are actually logged in.
+          example: 71303ce0-de89-496d-8270-6434589615e8
+        items:
+          type: array
+          description: Items purchased.
+          minItems: 1
+          items:
+            $ref: '#/components/schemas/PurchaseItem'
+        id:
+          type: string
+          description: >
+            The marketplace unique ID for the order.
+            This field ensures the event reporting is idempotent in case there is a network issue and the request is retried.
+            If there is no unique ID for orders on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: 0e06c899-b2cd-4e0d-b0de-8aefb4b6d0a0
+    PurchaseItem:
+      type: object
+      required:
+        - productId
+        - unitPrice
+      properties:
+        productId:
+          type: string
+          description: The marketplace ID of the product being purchased.
+          example: p_SA0238
+        resolvedBidId:
+          type: string
+          description: If known, the product's `resolvedBidId` if the consumer clicked on a promoted link before purchasing.
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+        quantity:
+          type: integer
+          minimum: 1
+          default: 1
+          description: Count of products purchased.
+          example: 2
+        unitPrice:
+          type: integer
+          minimum: 1
+          description: The price of a single item in minor currency units. For example, in the US (currency code "USD") the unit price is specified in cents.
+          example: 1295
+      example:
+        productId: p_SA0238
+        unitPrice: 1295
+        resolvedBidId: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+    EventsResponse:
+      type: object
+      properties:
+        impressions:
+          type: array
+          items:
+            type: object
+            properties:
+              impressionId:
```

Comment:
Several comments about this:
- could we call this `id` instead?
- cold/should we flatten this just like we do with the product ids in auctions?
- what's the purpose of this id? The fact that it cannot be used for anything is rather useless and considering they must send us now an event id, this id is not very useful.

I would remove it.

Reply:
Makes sense.